npm/metrics/namespaces.go

```diff
@@ -0,0 +1,16 @@
+package metrics
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// RecordNamespaceApplyTime adds an observation of namespace apply time for the specified apply mode.
+// The execution time is from the timer's start until now.
+func RecordNamespaceApplyTime(timer *Timer, mode ApplyMode) {
```

Comment:
Can we rename this func to RecordNamespaceApplyTime -> RecordNamespaceExecTime, same with other resource functions too, since this is not just apply instead it lso records delete and update